lib/kdump_utils.pm

```diff
@@ -34,8 +33,23 @@ sub install_kernel_debuginfo {
     # kernel-default(x86-64) = 5.3.18-lp152.26.2
     # kernel-default-base(x86-64) = 5.3.18-lp152.26.2.lp152.8.2.2
     # -> kernel-default-debuginfo-5.3.18-lp152.26.2
-    my $debuginfo = script_output('rpm -qf /boot/initrd-$(uname -r) --provides | awk \'match($0,/(kernel-.+)\(.+\) = (.+)/,m) {printf "%d %s-debuginfo-%s\n", length($0), m[1], m[2]}\' | sort -n | head -n1 | cut -d" " -f2-');
-    zypper_call("-v in $debuginfo", timeout => 4000);
+    return script_output('rpm -qf /boot/initrd-$(uname -r) --provides | awk \'match($0,/(kernel-.+)\(.+\) = (.+)/,m) {printf "%d %s-debuginfo-%s\n", length($0), m[1], m[2]}\' | sort -n | head -n1 | cut -d" " -f2-');
+}
+
+sub install_kernel_debuginfo {
```

Comment:
There are too many decisions in the function about transactional logic. I think would be better to create separate function `install__transactional_kernel_debuginfo`. Added if-else decreased readability of logic in this case.